pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/ColumnMetadata.java

```diff
@@ -88,6 +94,30 @@ public static ColumnMetadata fromPropertiesConfiguration(String column, Properti
     }
     builder.setPaddingCharacter(paddingCharacter);
 
+    // DERIVED_METRIC_TYPE property is used to check whether this field is derived or not
+    // ORIGIN_COLUMN property is used to indicate the origin field of this derived metric
+    String typeStr = config.getString(getKeyFor(column, DERIVED_METRIC_TYPE), null);
+    DerivedMetricType derivedMetricType = (typeStr == null)? null : DerivedMetricType.valueOf(typeStr.toUpperCase());
+
+    if (derivedMetricType != null) {
+      switch (derivedMetricType) {
+        case HLL:
+          try {
+            final int hllLog2m = config.getInt(V1Constants.MetadataKeys.StarTree.STAR_TREE_HLL_LOG2M);
```

Comment:
When there is no missing value for the key, the config.getInt(key) throws an exception, which is immediately caught by my code in several lines below.